File: app/src/main/java/com/cjmobileapps/quidditchplayersandroid/data/model/Player.kt
package com.cjmobileapps.quidditchplayersandroid.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID

data class Player(
    val id: UUID,
    val firstName: String,
    val lastName: String,
    val yearsPlayed: List<Int>,
    val favoriteSubject: String,
    val position: Int,
    val imageUrl: String,
    val house: HouseName
)

@Entity
data class PlayerEntity(
    @PrimaryKey val id: UUID,
    val firstName: String,
    val lastName: String,
    val yearsPlayed: List<Int>,
    val favoriteSubject: String,
    val position: String,
    val imageUrl: String,
    val house: HouseName
)

fun List<Player>.toPlayersEntities(positions: Map<Int, Position>): List<PlayerEntity> {
    return this.map { it.toPlayerEntity(positions) }
}

fun Player.toPlayerEntity(positions: Map<Int, Position>): PlayerEntity {
    return PlayerEntity(
        id = id,
        firstName = firstName,
        lastName = lastName,
        yearsPlayed = yearsPlayed,
        favoriteSubject = favoriteSubject,
        position = positions[this.position]?.positionName ?: "",
        imageUrl = imageUrl,
        house = house
    )
}

File: app/src/main/cpp/src/com/cjmobileapps/quidditchplayers/model/Player.cpp
#include "Player.h"
#include <string>
#include <vector>
#include <House.h>

namespace com::cjmobileapps::quidditchplayers::model {

    Player::Player(std::string id, std::string firstName, std::string lastName, std::vector<int> yearsPlayed,
                   std::string favoriteSubject, const int position, std::string imageUrl, const HouseName house)
        : id(std::move(id)), firstName(std::move(firstName)), lastName(std::move(lastName)),
          yearsPlayed(std::move(yearsPlayed)), favoriteSubject(std::move(favoriteSubject)),
          position(position), imageUrl(std::move(imageUrl)), house(house) {}

    std::string Player::getId() const {
        return id;
    }

    std::string Player::getFirstName() const {
        return firstName;
    }

    std::string Player::getLastName() const {
        return lastName;
    }

    std::vector<int> Player::getYearsPlayed() const {
        return yearsPlayed;
    }

    std::string Player::getFavoriteSubject() const {
        return favoriteSubject;
    }

    int Player::getPosition() const {
        return position;
    }

    std::string Player::getImageUrl() const {
        return imageUrl;
    }

    HouseName Player::getHouse() const {
        return house;
    }
}
